skills/hears.js

```diff
@@ -24,23 +23,38 @@ const extractSkills = (messageString) => {
 };
 
 
-const handler = (bot, message) => {
-
+const handler = async (bot, message) => {
   const messageContent = message.text.replace(LEARNING_KEY, ' ');
-  var skills = extractSkills(messageContent);
-
-  skills.forEach(skill => {
-    Skill.query()
-      .insert({ name: skill })
-      .then(res => bot.reply(message, `${skill} was added as a new skill!`),
-        err => {
-          if (!(err instanceof UniqueViolationError)) {
-            bot.reply(message, `Unable to add ${skill} as a skill :(`);
-          }
-          bot.reply(message, err);
+  const skills = extractSkills(messageContent);
+
+  if (skills.length) {
+    try {
+      const messageRecord = await Message.query().insertAndFetch({
```

Comment:
Hmm I hadn't considered this situation before, but this will currently not record any messages without a skill. i.e you can't say `thanks @tony for helping me`. Have you checked that this is what we want to enforce?